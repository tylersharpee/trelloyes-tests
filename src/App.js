import React from 'react';
import store from './store.js';
import List from './List';

function App(props) {
  const lists = props.store.lists.map(list => {
    return <List key={list.id} header={list.header} cards={list.cardIds.map(id => store.allCards[id])} />;
  })
  return (
    <main className="App">
      <header className="App-header">
        <h1>Trelloyes!</h1>
      </header>
      <div className="App-list">
        {lists}
      </div>
    </main>
  );
}

export default App;