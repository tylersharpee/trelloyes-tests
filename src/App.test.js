import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

//Snapshot Test
it ('renders the ui as expected', () => {
  const store = {
    lists: [
      {
        id: '1',
        header: 'First list',
        cardIds: [ 'a', 'b', 'e', 'f', 'g', 'j', 'l', 'm' ]
      }
    ],
    allCards: {
      'a': { id: 'a', title: 'First card', content: 'lorem ipsum' }
    }
  } 
  const tree = renderer
    .create(<App store={store} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

//Smoke Test
it ('render without crashing', () => {
  //create a DOM element to render the component
  const div = document.createElement('div');
  const store = {
    lists: [
      {
        id: '1',
        header: 'First list',
        cardIds: [ 'a', 'b', 'e', 'f', 'g', 'j', 'l', 'm' ]
      }
    ],
    allCards: {
      'a': { id: 'a', title: 'First card', content: 'lorem ipsum' }
    }
  } 
  //render the component
  ReactDOM.render(<App store={store} />, div);

  //clean up code
  ReactDOM.unmountComponentAtNode(div);
})