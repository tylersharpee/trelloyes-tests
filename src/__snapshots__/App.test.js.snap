// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ui as expected 1`] = `
<main
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Trelloyes!
    </h1>
  </header>
  <div
    className="App-list"
  >
    <section
      className="List"
    >
      <header
        className="List-header"
      >
        <h2>
          First list
        </h2>
      </header>
      <div
        className="List-cards"
      >
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            First card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Second card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Fifth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Sixth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Seventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Tenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Twelfth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Thirteenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </div>
    </section>
  </div>
</main>
`;
